### ISSUE TITLE
Multiple issues with InterFrame

### ISSUE THREAD

**Issue by Rast1234** [HomeOfVapourSynthEvolution/havsfunc#2]
Hello. I tried to post this in your doom9 thread, but the forum requires me to wait 5 days after registration...

I wanted to use your ported version of InterFrame on my Archlinux, successfully installed everything from AUR, but there were errors indicating that code passes wrong parameters to SVP methods. For example, SuperString is normally `{gpu:1}` which causes several exceptions:

1. SVP methods expect JSON-formatted string, so it should be `{"gpu":1}` (notice the double quotes). Without quotes, it throws `SVSuper: invalid 'params' syntax: * Line 1, Column 2  Missing '}' or object member name`.
2. After fixing this, i get an error `SVSuper: pel has to be 1 or 2 or 4`. Looks like `SVSuper()` and others do not use default parameters from SVP docs, so we need to set all values manually.
3. If `pel` value is 2 or 4, i get an error `SVSuper: can't ignore finest level with high pel value` which is fixed by adding `"full": True` to the parameters.
4. Same things also happen with `VectorsString` and `SmoothString`, because they are not JSON and SVP does not use default values for nonpresent parameters.

After digging all these things, i decided to refactor the code where you build `SuperString, VectorsString, SmoothString` to create dict object and then simply `json.dumps()`. Now the thing works for me!

However, if i set `GPU=True` (which causes `"gpu": 1` in SuperString), i get `SVSmoothFps: invalid super clip params` at the last step, `SmoothFps()`. I played with different parameters for all 3 methods with no luck.

So, i wanted to make a pull request with more pythonic and readable code of InterFrame, but i can't make it work with `GPU=True`. Do you have any ideas how to troubleshoot last exception? Or maybe you have SVP libs that work with your current code, and the only problem is my Archlinux with bleeding edge?

**Comment by HolyWu** [comment]
It's weird that you need to specify `"{"gpu":1}"` instead of `"{gpu:1}"`, because the latter works fine in both AviSynth and VapourSynth, at least on Windows. https://www.svp-team.com/wiki/Plugins:_SVPflow#Samples also shows the same usage. Therefore I have no idea if lib-linux has different behavior than lib-windows. Did you download SVPflow plugin from https://www.svp-team.com/wiki/Download#libs?

**Comment by Rast1234** [comment]
I installed everything from AUR (archlinux user repository). I will try it with vapoursynth and SVPFlow on windows and post back.

**Comment by Hinterwaeldlers** [comment]
Testing it with the lib https://www.svp-team.com/wiki/Download#libs and the libs delivered in the actual version of the installer worked both for me, without modifying the InterFrame code itself. Using GUI=True results in `SVSmoothFps: unable to init GPU-based renderer - code 65791` which sounds like a problem related to the support of the graphic card (got only the intel cpu ones). At least the things I've found while search for it (used error code 0x100ff as hex equivalent).
Tested on gentoo amd64. As I don't have access to windows I cannot test it there, too.

So does the problem still exist?
